plugins/a11y-keyboard/msd-a11y-keyboard-manager.c

```diff
@@ -80,6 +88,14 @@ G_DEFINE_TYPE_WITH_PRIVATE (MsdA11yKeyboardManager, msd_a11y_keyboard_manager, G
 
 static gpointer manager_object = NULL;
 
+static gboolean
+togglekeys_backend_enabled (MsdA11yKeyboardManager *manager,
+                            TogglekeysBackend       backend)
+{
+        return (g_settings_get_boolean (manager->priv->settings, "togglekeys-enable") &&
+                g_settings_get_enum (manager->priv->settings, "togglekeys-backend") == (gint) backend);
```

Comment:
You can merge both keys in the enum "togglekeys-backend", removing "togglekeys-enable" key.
```C
typedef enum
{
        TOGGLEKEYS_BACKEND_NONE,
        TOGGLEKEYS_BACKEND_XKB,
        TOGGLEKEYS_BACKEND_INTERNAL
} TogglekeysBackend;
```

Reply:
Well, this would mean that if the user disabled togglekeys, it would also lose the info on which backend was used.  IMO, it is not a good idea because I think it wouldn't be uncommon to enable or disable togglekeys, but changing backends shouldn't be something that happens often -- of ever after an initial setup maybe.
Also doing so would be incompatible with current configuration, which I don't think is legitimate in this case.

So although indeed for the piece of code you're mentioning it would make sense, I don't think it's actually a good idea to maintain best user experience and settings flexibility.